Applications/Games/Space Engineers/Steam/script.js

```diff
@@ -0,0 +1,38 @@
+include(["engines", "wine", "quick_script", "steam_script"]);
+include(["engines", "wine", "verbs", "dotnet461"]);
+include(["engines", "wine", "verbs", "vcrun2015"]);
+include(["engines", "wine", "verbs", "dxvk"]);
+include(["engines", "wine", "verbs", "xact"]);
+
+var installerImplementation = {
+    run: function () {
+        new SteamScript()
+            .name("Space Engineers")
+            .editor("Keen Software House")
+            .author("Zemogiter")
+            .appId(244850)
+            .wineVersion("4.0-rc3")
+            .wineDistribution("upstream")
+            .wineArchitecture("amd64")
+            .preInstall(function (wine, wizard) {
+                var operatingSystemFetcher = Bean("operatingSystemFetcher");
+                if (operatingSystemFetcher.fetchCurrentOperationSystem() !== "Linux")
```

Comment:
print the result of this method to see why it doesn't work

Reply:
I've added the print line above the `operatingSystemFetcher` variable but there is nothing related to this error in log file.

Reply:
On command line if your run from there?

Reply:
Where's the output you've added? Give it some prefix so that you can find it (e.g. "#### OS: ").

Reply:
You mean the print line?

Reply:
yes.

Reply:
```
print("Debug output");
var operatingSystemFetcher = Bean("operatingSystemFetcher");
if (operatingSystemFetcher.fetchCurrentOperationSystem() !== "Linux")
{
    throw "This game will not work under your OS due to MoltenVK not being implemented into Wine.".format("Space Engineers");
}
```

Reply:
```Javascript
print("Debug output: " + operatingSystemFetcher.fetchCurrentOperationSystem());
```

Reply:
```
javax.script.ScriptException: TypeError: Cannot read property "fetchCurrentOperationSystem" from undefined in <eval> at line number 22
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.throwAsScriptException(NashornScriptEngine.java:477)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:461)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:413)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:409)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.eval(NashornScriptEngine.java:162)
	at java.scripting/javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:264)
	at org.phoenicis.scripts/org.phoenicis.scripts.nashorn.NashornEngine.eval(NashornEngine.java:50)
	at org.phoenicis.scripts/org.phoenicis.scripts.nashorn.NashornScriptInterpreter.runScript(NashornScriptInterpreter.java:35)
	at org.phoenicis.scripts/org.phoenicis.scripts.interpreter.BackgroundScriptInterpreter.lambda$runScript$0(BackgroundScriptInterpreter.java:35)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: <eval>:22 TypeError: Cannot read property "fetchCurrentOperationSystem" from undefined
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ECMAErrors.error(ECMAErrors.java:57)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ECMAErrors.typeError(ECMAErrors.java:213)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ECMAErrors.typeError(ECMAErrors.java:185)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ECMAErrors.typeError(ECMAErrors.java:172)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.Undefined.get(Undefined.java:161)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$234$721AA$\^eval\_.run#L:21(<eval>:22)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$233$5556$\^eval\_$cu1$restOf.go([engines, wine, quick_script, steam_script]:141)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$72$400$\^eval\_.run(<eval>:13)
	at jdk.nashorn.javaadapters.org_phoenicis_scripts_Installer.run(Unknown Source)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$71$\^eval\_$cu1$restOf.:program(<eval>:46)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptFunctionData.invoke(ScriptFunctionData.java:655)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptFunction.invoke(ScriptFunction.java:513)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptRuntime.apply(ScriptRuntime.java:527)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:456)
	... 10 more
```

Reply:
Must be below:
```Javascript
var operatingSystemFetcher = Bean("operatingSystemFetcher");
```

Reply:
```
[INFO ] org.phoenicis.javafx.views.common.ThemeManager (l.74) - Setting theme to 'Standard theme'
[INFO ] org.phoenicis.javafx.views.common.ThemeManager (l.106) - Refreshing currently shown theme
[INFO ] org.phoenicis.javafx.views.common.ThemeManager (l.112) - Loading standard theme at 'jar:file:///usr/share/phoenicis-playonlinux/phoenicis-javafx-5.0-SNAPSHOT.jar!/org/phoenicis/javafx/themes/standard/main.css'
[INFO ] org.phoenicis.javafx.views.common.ThemeManager (l.121) - Loading WebView stylesheet at 'jar:file:///usr/share/phoenicis-playonlinux/phoenicis-javafx-5.0-SNAPSHOT.jar!/org/phoenicis/javafx/themes/standard/description.css'
[INFO ] org.phoenicis.repository.DefaultRepositoryManager (l.85) - Adding repositories: [org.phoenicis.repository.location.GitRepositoryLocation@55961b78[https://github.com/PhoenicisOrg/scripts,master], org.phoenicis.repository.location.LocalRepositoryLocation@1f42dc53[/home/jonasz/Pobrane/scripts-master]] at index 0
[INFO ] org.phoenicis.repository.types.MultipleRepository (l.46) - Fetching applications for: org.phoenicis.repository.types.MultipleRepository@c148224[repositories=[git-repository https://github.com/PhoenicisOrg/scripts, local folder: /home/jonasz/.Phoenicis/cache/git-331995911, branch: master, org.phoenicis.repository.types.LocalRepository@2d36dd6[repositorySource=file:/home/jonasz/Pobrane/scripts-master/,repositoryDirectory=/home/jonasz/Pobrane/scripts-master]]]
[INFO ] org.phoenicis.repository.types.GitRepository (l.76) - Begin fetching process of git-repository https://github.com/PhoenicisOrg/scripts, local folder: /home/jonasz/.Phoenicis/cache/git-331995911, branch: master
[INFO ] org.phoenicis.repository.types.LocalRepository (l.76) - Reading directory : /home/jonasz/Pobrane/scripts-master
[INFO ] org.phoenicis.repository.types.GitRepository (l.145) - Opening git-repository https://github.com/PhoenicisOrg/scripts, local folder: /home/jonasz/.Phoenicis/cache/git-331995911, branch: master
[INFO ] org.phoenicis.repository.types.GitRepository (l.149) - Pulling new commits from git-repository https://github.com/PhoenicisOrg/scripts, local folder: /home/jonasz/.Phoenicis/cache/git-331995911, branch: master
[INFO ] org.phoenicis.repository.types.LocalRepository (l.76) - Reading directory : /home/jonasz/.Phoenicis/cache/git-331995911
[INFO ] org.phoenicis.javafx.views.common.ThemeManager (l.106) - Refreshing currently shown theme
[INFO ] org.phoenicis.javafx.views.common.ThemeManager (l.112) - Loading standard theme at 'jar:file:///usr/share/phoenicis-playonlinux/phoenicis-javafx-5.0-SNAPSHOT.jar!/org/phoenicis/javafx/themes/standard/main.css'
[INFO ] org.phoenicis.javafx.views.common.ThemeManager (l.121) - Loading WebView stylesheet at 'jar:file:///usr/share/phoenicis-playonlinux/phoenicis-javafx-5.0-SNAPSHOT.jar!/org/phoenicis/javafx/themes/standard/description.css'

(java:20651): Gdk-WARNING **: 15:24:59.148: XSetErrorHandler() called with a GDK error trap pushed. Don't do that.
[INFO ] org.phoenicis.javafx.views.common.ThemeManager (l.106) - Refreshing currently shown theme
[INFO ] org.phoenicis.javafx.views.common.ThemeManager (l.112) - Loading standard theme at 'jar:file:///usr/share/phoenicis-playonlinux/phoenicis-javafx-5.0-SNAPSHOT.jar!/org/phoenicis/javafx/themes/standard/main.css'
[INFO ] org.phoenicis.javafx.views.common.ThemeManager (l.121) - Loading WebView stylesheet at 'jar:file:///usr/share/phoenicis-playonlinux/phoenicis-javafx-5.0-SNAPSHOT.jar!/org/phoenicis/javafx/themes/standard/description.css'
Warning: Nashorn engine is planned to be removed from a future JDK release
[WARN ] org.phoenicis.javafx.views.mainwindow.installations.dto.InstallationDTO (l.53) - Installation ID (Space Engineers_1546179907661) contains invalid characters, will remove them.
Debug output: Linux
```
The last line is interesting. The line returns my OS but despite the condition being false, the error message shows up.

Reply:
Can you try `!=` instead of `!==`? I'm not sure which datatype is really returned there. Maybe it is even required to use `operatingSystemFetcher.fetchCurrentOperationSystem().toString()`.

Reply:
Solved.

Reply:
Ok, then you can remove the debug print again.